### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by dlongley**

### CODE REVIEW

File: Cipher.js [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+   *   `recommended` to use the latest recommended algorithm when encrypting.
+   *
+   * @return {Cipher}.
+   */
+  constructor({version = 'recommended'} = {}) {
+    if(typeof version !== 'string') {
+      throw new TypeError('"version" must be a string.');
+    }
+    if(!VERSIONS.includes(version)) {
+      throw new Error(`Unsupported version "${version}"`);
```
Comment (mattcollier) [inline Cipher.js]: Needs a period.

File: Cipher.js [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+   * generated. If a new CEK is to be generated, then all other recipients must
+   * reference an accessible Diffie-Hellman public key so that this method can
+   * wrap the CEK for them.
+   *
+   * @param {Uint8Array|String} data the data to encrypt.
+   * @param {Object} kek a key encryption key API with `id`, `wrap`, and
+   *   `unwrap`.
+   * @param {Array} recipients an optional array of recipients for the
+   *   encrypted content.
+   * @param {String} wrappedCek an optional base64url-encoded CEK.
```
Comment (mattcollier) [inline Cipher.js]:
```suggestion
   * @param {String} [wrappedCek] a base64url-encoded CEK.
```

File: DataHub.js [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+  }
+
+  /**
+   * Ensures that future documents inserted or updated using this DataHub
+   * instance will be indexed according to the given attribute, provided that
+   * they contain that attribute.
+   *
+   * @param {Array|Object} attribute the attribute name or an array of
+   *   attribute names.
+   * @param {Boolean} unique `true` if attribute values should be considered
```
Comment (mattcollier) [inline DataHub.js]:
```suggestion
   * @param {Boolean} [unique=false] `true` if attribute values should be considered
```
And update verbiage.

**Review by mattcollier — APPROVED** [review]
Just a few nits.  Congratulations on getting this one done!

File: Cipher.js (hunk 1 above)
Comment (dlongley) [reply Cipher.js]: Fixed

**Comment by dlongley** [comment]
@mattcollier, thanks!

File: Cipher.js [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+};
+
+export class Cipher {
+  /**
+   * Creates a new Cipher instance that can be used to encrypt or decrypt
+   * data. A version must be supplied for encrypting data; the version
+   * indicates whether a FIPS-compliant algorithm or the latest recommended
+   * algorithm will be used.
+   *
+   * @param {String} version `fips` to use a FIPS-compliant algorithm,
```
Comment (davidlehn) [inline Cipher.js]: looks like version is optional

File: Cipher.js [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+   * wrapped content encryption key (CEK) can be found in `recipients` for
+   * `kek`, then it will be unwrapped and reused. Otherwise, a new CEK will be
+   * generated. If a new CEK is to be generated, then all other recipients must
+   * reference an accessible Diffie-Hellman public key so that this method can
+   * wrap the CEK for them.
+   *
+   * @param {Uint8Array|String} data the data to encrypt.
+   * @param {Object} kek a key encryption key API with `id`, `wrap`, and
+   *   `unwrap`.
+   * @param {Array} recipients an optional array of recipients for the
```
Comment (davidlehn) [inline Cipher.js]: optional

File: DataHub.js [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+import axios from 'axios';
+import {Cipher} from './Cipher.js';
+import {IndexHelper} from './IndexHelper.js';
+
+export class DataHub {
+  /**
+   * Creates a new DataHub instance. The storage for the data hub must already
+   * exist and have an HTTPS API at the given `baseUrl`.
+   *
+   * @param {String} baseUrl an optional endpoint for the data hub storage.
```
Comment (davidlehn) [inline DataHub.js]: optional

File: Cipher.js (hunk 5 above)
Comment (dlongley) [reply Cipher.js]: Fixed

File: Cipher.js (hunk 4 above)
Comment (dlongley) [reply Cipher.js]: Fixed

File: DataHub.js (hunk 6 above)
Comment (dlongley) [reply DataHub.js]: Fixed

File: DataHubService.js [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+      base: '/data-hubs'
+    }
+  } = {}) {
+    this.config = {urls};
+  }
+
+  /**
+   * @method create
+   * @param {Object} options
+   * @param {string} options.baseUrl - the base baseUrl
```
Comment (davidlehn) [inline DataHubService.js]: baseUrl optional and called url.

File: DataHubService.js [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+   * @param {string} options.config - the data hub's configuration
+   * @description returns the configuration for the newly created data hub.
```
Comment (davidlehn) [inline DataHubService.js]: Could untag descriptions and put at top here and below.

File: DataHubService.js (hunk 7 above)
Comment (davidlehn) [reply DataHubService.js]: It's baseUrl elsewhere, maybe the code in this function should be renamed vs param.

File: LICENSE.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
+Bedrock Non-Commercial License v1.0
+===================================
+
+Copyright (c) 2011-2018 Digital Bazaar, Inc.
```
Comment (davidlehn) [inline LICENSE.md]: 2019

File: DataHubService.js (hunk 7 above)
Comment (dlongley) [reply DataHubService.js]: Fixed...enough

File: DataHubService.js (hunk 8 above)
Comment (dlongley) [reply DataHubService.js]: Fixed

File: LICENSE.md (hunk 9 above)
Comment (dlongley) [reply LICENSE.md]: Fixed

**Pull request merged**